### ISSUE TITLE
False positive: sentence-per-line detection with footnote fails

### ISSUE THREAD

**Issue by Gabgobie** [rvben/rumdl#716]
### rumdl version

0.2.28

### Installation method

`uv`

### Bug description

`[MD013] Paragraph should have one sentence per line (found 3 sentences across 4 lines) [*]`

This was caused by the dot (`.`) of the first line being followed by a footnote marker `[^1]`.

### Steps to reproduce

1. Configure `rumdl` for [`sentence-per-line` formatting](https://rumdl.dev/md013/?h=line+length#sentence-per-line-without-line-length-warnings)
1. Create a file `test.md`
   ```markdown
   Sentence one needs an annotation.[^1]
   Sentence two works fine.
   Sentence three as well.
   Four too.

   [^1]: Annotation for sentence one.
   ```
1. Run `rumdl check test.md`
1. Linter complains

### Expected behavior

I'd expect `rumdl` to recognize that the sentence ended.

### Actual behavior

`rumdl` thinks that the sentence didn't end because the line didn't end on a dot.

### Configuration

```toml
# pyproject.toml

# MD013 - Keep lines short for better readability
[tool.rumdl.MD013]
line-length = 0
paragraphs = false
code-blocks = true
tables = true
reflow = true
reflow-mode = "sentence-per-line"
```

### Operating system

Windows

**Comment by Gabgobie** [comment]
Appreciate you getting to this so quickly!

If you don't mind a follow up question.

```markdown
- list item before
- Google Calendar
  (Can't we get rid of this dependency? - I don't really see the need)
- list item after
```

This throws `Paragraph should have one sentence per line (found 2 sentences across 2 lines)`. How would I correctly format this or is this another bug?

Originally, the `Google Calendar [...]` item was one line (and rejected)

---

Edit: Maybe this example should get its own bug report. `rumdl fmt` incorrectly turns it into a sub item

After `fmt`:

```markdown
- list item before
- Google Calendar (Can't we get rid of this dependency?
  - I don't really see the need)
- list item after
```

**Comment by rvben** [comment]
Fixed in v0.2.32. The sentence-boundary detector only inspected the single character after sentence-ending punctuation, so `.[^1]` never registered as a sentence end and `fmt` merged the annotated sentence into the next one. One or more consecutive footnote references immediately after the punctuation now count as the sentence end, and the references stay attached to the sentence they annotate. Bare brackets like `.[1]` intentionally still continue the sentence. The list/parenthetical reflow behavior from your follow-up comment looks like a separate bug; if you can open it as its own issue with the exact input, I'll pick it up.
